tests/plugins/test_add_filesystem.py

```diff
@@ -295,6 +300,13 @@ def test_add_filesystem_plugin_generated(workflow, build_dir, scratch, dockerfil
     assert msg in caplog.text
     assert workflow.build_dir.any_platform.dockerfile.content == expected_dockerfile
 
+    # Verify custom parent image is marked as scratch
+    for pullable, local_img in runner.workflow.data.dockerfile_images.items():
+        if base_image_is_custom(str(pullable)):
+            assert local_img == ImageName.parse(SCRATCH_FROM)
```

Comment:
Ah, I don't think this will actually work. I think `ImageName.parse('scratch')` results in `scratch:latest`